blaze/expr/broadcast.py

```diff
@@ -65,8 +65,14 @@ def _name(self):
 
     @property
     def _full_expr(self):
-        return self._scalar_expr._subs(dict(zip(self._scalars,
-                                                self._children)))
+        return self._scalar_expr._subs(
+            dict(zip(self._scalars, self._children))
+        )
+
+    @property
+    def _args(self):
+        # XXX: is this a hack?
```

Comment:
we need because otherwise there's no path from a broadcast expression to the true leaf because for the following expression `t.amount + 1`, `Broadcast`'s `_args` look like this `_children=(t,), _scalars=(t,), _scalar_expr=t.amount + 1)` and the `path` algorithm will traverse an expression until we've reached the end of the path and in this case we are traversing to a leaf which doesn't appear in `Broadcast._traverse()` because everything is scalar-ified